backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/OutputGenerator.java

```diff
@@ -119,10 +119,29 @@ protected LicenseInfoParsingResult mergeLicenseInfoParsingResults(LicenseInfoPar
                 !getComponentLongName(r1).equals(getComponentLongName(r2))){
             throw new IllegalArgumentException("Only successful parsing results for the same release can be merged");
         }
+
         LicenseInfoParsingResult r = new LicenseInfoParsingResult(r1);
-        r.getLicenseInfo().getLicenseNamesWithTexts().addAll(r2.getLicenseInfo().getLicenseNamesWithTexts());
-        r.getLicenseInfo().getCopyrights().addAll(r2.getLicenseInfo().getCopyrights());
-        r.getLicenseInfo().getFilenames().addAll(r2.getLicenseInfo().getFilenames());
+        if (r2.getLicenseInfo().isSetLicenseNamesWithTexts()) {
```

Comment:
I would posit that this option is easier to read:
```java
        Optional.of(r).map(LicenseInfoParsingResult::getLicenseInfo).map(LicenseInfo::getLicenseNamesWithTexts)
                .ifPresent(set -> set.addAll(Optional.of(r2)
                        .map(LicenseInfoParsingResult::getLicenseInfo)
                        .map(LicenseInfo::getLicenseNamesWithTexts)
                        .orElse(Collections.emptySet())));
```

What do you think?

Reply:
Hmmm I had a similar solution with a lambda function to solve that problem.
But I thought its not really readable.
I would say that your solution is not more clearly. :)

Reply:
let's get a second opinion. @maxhbr, what do you say?

Reply:
@alexbrdn : I see two problems with your code:
- it contains duplicated code: `.map(LicenseInfoParsingResult::getLicenseInfo).map(LicenseInfo::getLicenseNamesWithTexts)`
- it only works if `r.getLicenseInfo().isSetLicenseNamesWithTexts() == true` (otherwise nothing is done, due to the first `ifPresent`)

I would write
```
r.getLicenseInfo().setLicenseNamesWithTexts().addAll(
    nullToEmptyCollection(
        r2.getLicenseInfo().getLicenseNamesWithTexts()));
```
But that also expects that `r.getLicenseInfo().isSetLicenseNamesWithTexts() == true`.

Reply:
r.getLicenseInfo().setLicenseNamesWithTexts().addAll(
    nullToEmptyCollection(
        r2.getLicenseInfo().getLicenseNamesWithTexts()));
This will create an empty collection in case of null.
Why we should create a empty collection if the origin is null.

Yes both of the LicenseInfo objects can be null
`r.getLicenseInfo().getXXX` and `r2.getLicenseInfo().getXXX` can be null.

- 1. I am verify if i need to merge something `if (r2.getLicenseInfo().isSetLicenseNamesWithTexts())`
- 2. Verify If the origin object == null (set list to origin)
- 3. If the origin object != null just add the items which we want to merge